kernel/kernel-pdfgenerator/src/main/java/io/mosip/kernel/pdfgenerator/util/FontPdfRendererBuilder.java

```diff
@@ -0,0 +1,54 @@
+package io.mosip.kernel.pdfgenerator.util;
+import com.openhtmltopdf.pdfboxout.PdfRendererBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+public class FontPdfRendererBuilder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FontPdfRendererBuilder.class);
+
+    private static PdfRendererBuilder builderInstance;
+
+    private FontPdfRendererBuilder() {
+        // Private constructor to enforce singleton
+    }
+
+    public static synchronized PdfRendererBuilder getBuilder() throws IOException {
+        if (builderInstance == null) {
+            builderInstance = new PdfRendererBuilder();
+            initializeFonts(builderInstance);
+        }
+        return builderInstance;
+    }
+
+    private static void initializeFonts(PdfRendererBuilder builder) throws IOException {
+        PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+        Resource[] resources = resolver.getResources("classpath:/pdf-generator/*.ttf");
```

Comment:
can we take this path from the configuration? and keep the current hardcoded value as the default value.

Reply:
taken care